File: python-altair/plot.py
import pandas as pd
import altair as alt

data = pd.read_csv('../penglings.csv')

color_domain = ['Adelie', 'Chinstrap', 'Gentoo']
color_range = ['orange', '#A550EF', 'darkcyan']

alt.Chart(data).mark_circle(size=60).encode(
    x=alt.X('flipper_length_mm', title='Flipper Length (mm)', scale=alt.Scale(zero=False)),
    y=alt.Y('body_mass_g', title='Body Mass (g)', scale=alt.Scale(zero=False)),
    color=alt.Color('species', scale=alt.Scale(domain=color_domain, range=color_range)),
    size='bill_length_mm',
).configure_view( # AI Usage Note: Can you mimic the darkgrid style of seaborn/matplotlib?
    fill="#EBEBEB",
    stroke=None
).configure_axis(
    grid=True,
    gridColor="white",
    domain=False,
    ticks=False
).configure_mark(opacity=0.8).properties(width=600, height=400).save('altair-scatterplot.png', scale_factor=3)

File: r-ggplot2/Plot.R
penglings <- read.csv("../penglings.csv")

head(penglings)

library(ggplot2)

ggplot(penglings, aes(x=flipper_length_mm, y=body_mass_g, size=bill_length_mm, color=species)) + 
  geom_point(alpha=0.8) +
  theme_grey() +
  scale_color_manual(values = c(
    Adelie = "orange",
    Chinstrap = "purple",
    Gentoo = "darkcyan"
  )) +
  labs(x = "Flipper Length (mm)", y = "Body Mass (g)") +
  guides(
    size = guide_legend(order = 1),
    color = guide_legend(order = 2)
  )
